### PR TITLE
added support for custom gstreamer pipeline

### PR DESCRIPTION
**Pull request by enricomarchesin**
Users can now supply a fully customized gstreamer pipeline, using both a command line option (`-g`) and an environment variable (`GST_PIPELINE`).

Many interesting opportunities open, for example the possibility to stream a file without transcoding using a command like this:

```bash
docker run --rm -e INPUT="/tmp/video.mp4" -e RTSP_PORT=18554 -e "filesrc location=/tmp/video.mp4 ! qtdemux ! video/x-h264 ! rtph264pay pt=96 name=pay0" -v "/path/to/your/video:/tmp/video.mp4" -p 18554:18554 ullaakut/rtspatt
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,9 +48,10 @@ docker run --rm \
       [-p rtsp_password] \
       [-s rtsp_resolution] \
       [-f rtsp_framerate] \
+      [-i input]
       [-t] \
       [-d] \
-      [-i input]
+      [-g] gstreamer_pipeline \
```
Comment (Ullaakut) [inline README.md]: formatting should be `[-g gstreamer_pipeline]`

File: src/parsing.cpp [hunk 2]
```diff
@@ -97,13 +97,19 @@ void parse_env(std::shared_ptr<t_config> &config) {
...
+    config->pipeline = pipeline;
+  }
 }
 
 // Overwrite default parameters via cmd line
 bool parse_args(std::shared_ptr<t_config> &config, int argc, char **argv) {
   int c;
   opterr = 0;
-  while ((c = getopt(argc, argv, "r:u:l:p:b:f:s:i:ht")) != -1) {
+  while ((c = getopt(argc, argv, "r:u:l:p:b:f:s:i:htg:")) != -1) {
```
Comment (Ullaakut) [inline src/parsing.cpp]: please move the `g:` before `ht` (it's a tiny detail but it makes it easier to read when arguments are separated by whether or not they need a parameter

File: README.md (hunk 1 above)
Comment (enricomarchesin) [reply README.md]: ☺️

File: src/parsing.cpp (hunk 2 above)
Comment (enricomarchesin) [reply src/parsing.cpp]: make sense!

**Comment by enricomarchesin** [comment]
Thanks to you @Ullaakut for the great work done here: you've build a very handy tool! 👏

**Pull request merged**